### PR TITLE
ROGER-1089:Update nginx auth request to support request body

### PR DESCRIPTION
**Pull request by 04msambit**
Made changes to enable nginx to support forwarding of request body to AAAD.

Test:
1 - AAAD able to log body from request.
2 - AAAD able to throw back error based on body.

The above tests were carried out on localmesos01 for both marathon and chronos framework.

@amitbose327 Please have a look.

### CODE REVIEW

File: ansible/roles/nginx/tasks/main.yml [hunk 1]
```diff
@@ -54,6 +54,13 @@
       - install
       - nginx
 
+  - name: Copy Lua Auth Script file
```
Comment (amitbose327) [inline ansible/roles/nginx/tasks/main.yml]: Might be good to add a remove auth_script.lua task, when nginx_auth_enable is not defined or not true.

File: ansible/roles/nginx/files/auth_script.lua [hunk 2]
```diff
@@ -0,0 +1,18 @@
+ngx.req.read_body()
```
Comment (ankanm) [inline ansible/roles/nginx/files/auth_script.lua]: Can you please rename this to `auth_request.lua`?

File: ansible/roles/nginx/tasks/main.yml [hunk 3]
```diff
@@ -1,18 +1,18 @@
 ---
 
-  - name: Download nginx release
+  - name: Download openresty/nginx bundle
```
Comment (ankanm) [inline ansible/roles/nginx/tasks/main.yml]: What happens to the current nginx installation (if any)? Is there a need to remove that first (or perhaps manually)?

File: ansible/roles/nginx/templates/rogeros_auth_proxy.conf.j2 [hunk 4]
```diff
@@ -22,30 +22,30 @@ server {
...
-		# goes to another method if basic http auth fails
-		# in our case, it will always go to the aaadaemon
-		# because http credential files is empty
-		satisfy all;
-		auth_basic "Restricted";
-		auth_basic_user_file {{nginx_conf_dir}}/{{nginx_basic_auth_file_name}};
-		auth_request /auth-proxy;
+                # goes to another method if basic http auth fails
+                # in our case, it will always go to the aaadaemon
+                # because http credential files is empty
```
Comment (ankanm) [inline ansible/roles/nginx/templates/rogeros_auth_proxy.conf.j2]: The above 3 comment lines appears to be no longer valid (unrelated to this story.) Best to remove them if so.

File: ansible/roles/nginx/tasks/main.yml (hunk 1 above)
Comment (04msambit) [inline ansible/roles/nginx/tasks/main.yml]: Added task to remove the auth_request.lua when nginx_auth_enable is not defined or not true.

File: ansible/roles/nginx/files/auth_script.lua (hunk 2 above)
Comment (04msambit) [inline ansible/roles/nginx/files/auth_script.lua]: Renamed to auth_request.lua.

File: ansible/roles/nginx/tasks/main.yml (hunk 3 above)
Comment (04msambit) [inline ansible/roles/nginx/tasks/main.yml]: Yes , we need to remove any older existing nginx. Added steps/tasks to remove the old existing nginx ( /usr/local/nginx ).

File: ansible/roles/nginx/templates/rogeros_auth_proxy.conf.j2 (hunk 4 above)
Comment (04msambit) [inline ansible/roles/nginx/templates/rogeros_auth_proxy.conf.j2]: Removed.

File: ansible/roles/nginx/tasks/main.yml [hunk 5]
```diff
@@ -1,6 +1,12 @@
 ---
+  - name: Remove any existing NGINX under "/usr/local" when nginx_auth_enable is true
+    when: nginx_auth_enable is defined and nginx_auth_enable|bool
```
Comment (ankanm) [inline ansible/roles/nginx/tasks/main.yml]: Shouldn't this be removed either way (i.e., whether or not `nginx_auth_enable` is true)?

**Comment by amitbose327** [comment]
LGTM

**Comment by ankanm** [comment]
LGTM

**Pull request merged**